### DISCUSSION TITLE
Any plans for RP2350?

### DISCUSSION THREAD

**Discussion by hrvach** [sekigon-gonnoc/Pico-PIO-USB#134] [General]
First - thank you for an AMAZING project! This helped spawn so many other projects it's incredible...

There is a new and improved Pico 2 board featuring RP2350, mostly same price and even better PIO capabilities. Any plans in the roadmap to extend support for that one?

Thank you!

**Comment by arturo182** [comment]
I've been using this library on the RP2350 without much issue, but only on GPIOs up to 32, see #133

**Reply by hrvach** [reply]
That's amazing news, thank you! I'll try it out :)

**Reply by arturo182** [reply]
Make sure to use 0.6.1 as it has fixes for the new Pico SDK 2.0.0, and a bug introduced in 0.6.0

**Comment by gritd** [comment]
I've got a waveshare 2350 USB-A board which has DP / DM on resp. 12 and 13, now I think I discovered the ports can be set in pio_usb_configuration.h, but it does not really have any effect on recognising the USB device on these ports. What's the proper way to shift hosted usb ports to another io location? In their documentation I read #define HOST_PIN_DP   12 but that does not seem to match with the current lib config.

**Reply by sonik-br** [reply]
Try removing the two pull up resistors from DP/DM pins.
I've modded a 2350 USB-A this way and it started to work with some usb devices (but not all).

<img width="644" height="320" alt="image" src="https://github.com/user-attachments/assets/aaddb5cf-4a9e-4bdb-951b-594fc0ed477d" />
